NuGet.Config

```diff
@@ -45,27 +46,29 @@
       <package pattern="xunit.*" />
       <package pattern="xunitxml.testlogger" />
     </packageSource>
-    <packageSource key = "dotnet-eng">
+    <packageSource key="dotnet-eng">
       <package pattern="Microsoft.DotNet.*" />
     </packageSource>
-    <packageSource key = "nuget-build">
+    <packageSource key="nuget-build">
       <package pattern="nuget.client.endtoend.testdata" />
       <package pattern="nugetvalidator" />
     </packageSource>
-    <packageSource key = "vside">
+    <packageSource key="vside">
       <package pattern="envdte" />
       <package pattern="envdte100" />
       <package pattern="envdte80" />
       <package pattern="envdte90" />
       <package pattern="envdte90a" />
+      <package pattern="Microsoft.Build.Framework"/>
       <package pattern="Microsoft.DataAI.NuGetRecommender.Contracts" />
       <package pattern="microsoft.internal.visualstudio.*" />
       <package pattern="Microsoft.Internal.VisualStudio.Shell.Framework" />
+      <package pattern="Microsoft.NET.StringTools" />
```

Comment:
This package also exists on nuget.org https://www.nuget.org/packages/Microsoft.NET.StringTools

why do we need to get it from vside?

Reply:
So for both of your comments, I'm getting a NU1603 error. So the version isn't available on dotnet-public. Is there a way to add it there? 


but Microsoft.Build.Framework 17.10.0 was not found. Microsoft.Build.Framework 17.10.4 was resolved instead.

Reply:
It's not exactly the same thing, but here's an example, that I learned about just today, of how us changing MSBuild version numbers can have unintended consequences: https://github.com/dotnet/docs/pull/42393#discussion_r1746303334

If this Microsoft.Build.Framework or Microsoft.NET.StringTools version that is not on nuget.org ends up accidentally as a dependency of any of our own packages (which we do publish to nuget.org), then any customer who uses our package will then end up with an NU1603. We've had customers (once internal, once external) complain to us about this, and I've fixed it twice, hence why I feel strongly that any packages that exist on nuget.org, we should only use from the dotnet-public feed.

The dotnet-public feed already syncs all Microsoft owned packages from nuget.org, so if it's not already on dotnet-public then it's not on nuget.org.  You could ask the package owner if they're willing to publish that specific version (17.10.0). Both of these packages happen to be owned by the MSBuild team.

Alternatively, you can ask the owner of the package (Microsoft.VisualStudio.Shell.Framework) who lists that dependency to use a version that is public. But then they'd need to publish a new version of their package to have that updated dependency version. And whatever package listed that package as a dependency needs to update, and so up, until the package that we use is updated.

A last option is to add a `PackageReference` on the packages that are getting reported in the NU1603 message. I think by making it a direct dependency of the project, that should prevent the dependency resolver from trying to download the other version that's a transitive dependency.  But if the transitive version is a higher version that what we list in our CPM versions file, then we'll get a package downgrade error. And for Microsoft.Build.Framework, upgrading to a newer version isn't an option because of that issue reported in the dotnet docs today.

Ideally, the Markdown.Platform package wouldn't have both the implementation and API contracts in the same package, and that way transitive dependencies of its implementation (MSBuild's packages) wouldn't even come up in our restore graph.  This is exactly how NuGet's VS extensibility APIs work (NuGet.VisualStudio and NuGet.VisualStudio.Contracts, they're API interfaces only, no implementation, with minimal dependencies).  Or, if they removed their specific package dependencies, and depened on Microsoft.VisualStudio.SDK instead, then that would be easier for us to override.

So, unfortunately there might not be a simple solution ☹️

Reply:
Hmm okay I upgraded to the latest stable version of the platform tool and updated the MicrosoftBuildVersion to 17.11.4

Reply:
Oh wait, you just said we can't upgrade Microsoft.Build.Framework 😔

Reply:
Okay so leaving it back to that preview version and added a PackageReference to Microsoft.Build.Framework and Microsoft.NET.StringTools, so for now it works fine. 

I'll want to upgrade this package eventually, but the new version depends on `Microsoft.VisualStudio.ScriptedHost` which isn't hosted on a public feed right now. If we can get that published to a public feed that we can revisit this issue then.